app/models/miq_cockpit_ws_worker/runner.rb

```diff
@@ -0,0 +1,177 @@
+  end
+
+  def start_cockpit_ws_process
+    @pid, @stdout, @stderr = cockpit_ws_run
+  rescue => err
+    _log.error("#{log_prefix} cockpit-ws Process aborted because [#{err.message}]")
+    _log.log_backtrace(err)
+  end
+
+  def stop_cockpit_ws
+    if cockpit_ws_alive?
+      _log.info("#{log_prefix} Shutting down cockpit-ws process...pid=#{@pid}")
+      stop_cockpit_ws_process
+      _log.info("#{log_prefix} Shutting down cockpit-ws process...Complete")
+    end
+  end
+
+  def stop_cockpit_ws_process
+    return unless @pid
+    Process.kill("TERM", @pid)
+    wait_on_cockpit_ws
+  end
+
+  # Waits for a cockpit-ws process to stop.  The process is expected to be
+  #   in the act of shutting down, and thus it will wait 5 minutes
+  #   before issuing a kill.
+  def wait_on_cockpit_ws(pid)
+    pid ||= @pid
+    # TODO: Use Process.waitpid or one of its async variants
+    begin
+      Timeout.timeout(5.minutes.to_i) do
+        loop do
+          break unless process_alive?(pid)
+          sleep 1
+          heartbeat
+        end
+      end
+    rescue Timeout::Error
+      _log.info("#{log_prefix} Killing cockpit-ws process with pid=#{pid}")
+      Process.kill(9, pid)
+    end
+
+    begin
+      _, status = Process.waitpid2(pid)
+      _log.info("#{log_prefix} cockpit-ws Process with pid=#{pid} exited with a status=#{status}")
+    rescue Errno::ECHILD
+      _log.info("#{log_prefix} cockpit-ws Process with pid=#{pid} exited")
+    end
+
+    reset_process_info
+  end
+
+  def reset_process_info
+    @pid = @stdout = @stderr = nil
+  end
+
+  def find_cockpit_ws_processes
+    find_pids(/cockpit-ws --port/)
+  end
+
+  def stop_active_cockpit_ws_processes
+    find_cockpit_ws_processes.each do |pid|
+      _log.info("#{log_prefix} Killing active cockpit-ws process with pid=#{pid}")
+      Process.kill("TERM", pid)
+      wait_on_cockpit_ws(pid)
+    end
+  end
+
+  def cockpit_ws_alive?
+    if process_alive?(@pid)
+      true
+    else
+      reset_process_info
+      false
+    end
+  end
+
+  def process_alive?(pid)
+    return false if pid.nil?
+    process_struct = Sys::ProcTable.ps(pid)
+    return false if process_struct.nil?
+    return true if process_struct.state != :Z
+
+    if pid_state == :Z
+      zombie_pid, status = Process.waitpid2(pid)
+      _log.info("#{log_prefix} cockpit-ws Process with pid=#{zombie_pid} exited with a status=#{status}")
+    end
+
+    _log.info("#{log_prefix} cockpit-ws Process with pid=#{pid} is not alive pid_state=#{pid_state}")
+    false
+  end
+
+  def cockpit_ws_run
+    _log.info("#{log_prefix} cockpit-ws process starting")
+
+    opts = @worker_settings
+    cockpit_ws = MiqCockpit::WS.new(opts)
+    cockpit_ws.save_config
+
+    require 'open4'
+    env = {
+      "XDG_CONFIG_DIRS" => cockpit_ws.config_dir,
+      "DRB_URI"         => @drb_uri
+    }
+    pid, stdin, stdout, stderr = Open4.popen4(env, *cockpit_ws.command(BINDING_ADDRESS))
```

Comment:
Is there a reason we're using Open4 here instead of [AwesomeSpawn](https://github.com/ManageIQ/awesome_spawn) for process launching?  I'm just curious.  Maybe because it's a long running process?